utcp.c

```diff
@@ -1042,6 +1053,7 @@ ssize_t utcp_recv(struct utcp *utcp, const void *data, size_t len) {
 
 	// 3. Check incoming data for acceptable seqno and update connection timer
 
+	size_t datalen = len;
```

Comment:
@a-krebs len is the received packet data len while datalen is the remaining length of the data pointer to process after the overlapping is cut and data pointer adjusted

the received packet data len still is used for log output on rejection and for FIN check (hdr.seq + len == c->rcv.nxt) and sendatleastone ack check which previously got messed